tests/CacheCompat/CommonCache.Test.MsalNode/NodeScript/index.js

```diff
@@ -0,0 +1,188 @@
+﻿/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+
+/* To run this locally, start with `node index.js --inputPath c:\\temp\\input.json` where input.json contains
+
+{
+    "Scope": "https://graph.microsoft.com/user.read",
+    "CacheFilePath": "C:\\Users\\bogavril\\AppData\\Local\\Temp\\adalcachecompattestdata\\msalCacheV3.binx",
+    "LabUserDatas": [
+        {
+            "Upn": "liu.kang@bogavrilltd.onmicrosoft.com",
+            "Password": "****",
+            "ClientId": "1d18b3b0-251b-4714-a02a-9956cec86c2d",
+            "TenantId": "49f548d0-12b7-4169-a390-bb5304d24462",
+            "Authority": "https://login.microsoftonline.com/49f548d0-12b7-4169-a390-bb5304d24462/"
+        }
+    ],
+    "ResultsFilePath": "C:\\Users\\bogavril\\AppData\\Local\\Temp\\adalcachecompattestdata\\msal_python_results.json",
+    "StorageType": 4
+}
+
+Example output (should go in ResultsFilePath):
+
+{
+    "ErrorMessage": "", 
+    "IsError": false, 
+    "Results": [
+        {
+            "AuthResultUpn": "liu.kang@bogavrilLTD.onmicrosoft.com", 
+            "IsAuthResultFromCache": false, 
+            "LabUserUpn": "liu.kang@bogavrilltd.onmicrosoft.com"
+        }
+    ]
+}
+
+*/
+
+
+var msal = require("@azure/msal-node");
+const { Console } = require("console");
+const { promises: fs } = require("fs");
+
+async function readTestInputAsync() {
+
+    if (process.argv.length < 4) {
+        console.error("No arguments given. Expecting an input file path");
+        throw "No arguments given. Expecting an input file path.";
+    }
+    let inputFile = process.argv[3];
+
+    console.log("Reading input file: " + inputFile);
+
+    let rawInputData = await fs.readFile(inputFile);
+    let testInput = JSON.parse(rawInputData);
+    console.log("CacheFilePath: " + testInput.CacheFilePath);
+
+    return testInput;
+}
+
+async function createPCAAsync(testInput, userIndex) {
+    const cachePath = testInput.CacheFilePath;
+
+
+    const beforeCacheAccess = async (cacheContext) => {
+        const fileExists = async path => !!(await fs.stat(path).catch(e => false));
```

Comment:
fs operations return a specific error core if file does not exist. You could compare e.code == "ENOENT" if you want to specifically check for this condition, instead of just caching all exceptions.

Reply:
yeah, I copied it from SO and like how concise it is